driver/mseloss_driver.hpp

```diff
@@ -0,0 +1,732 @@
+        size_t TGidx = get5DIndexAt<size_t>(TG_tv, n0, n1, n2, n3, n4);
+
+        Tref grad =
+            static_cast<Tgpu>(2.0f) * (input[Iidx] - target[Tidx]) / divisor * output[O_tv.offset];
+
+        if(input_grad != nullptr)
+        {
+            input_grad[IGidx] = grad;
+        }
+
+        if(target_grad != nullptr)
+        {
+            target_grad[TGidx] = -grad;
+        }
+
+        ++gid;
+    }
+}
+
+template <typename Tgpu, typename Tref>
+class MSELossDriver : public Driver
+{
+
+public:
+    MSELossDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputDesc);
+        miopenCreateTensorDescriptor(&targetDesc);
+        miopenCreateTensorDescriptor(&outputDesc);
+
+        miopenCreateTensorDescriptor(&inputGradDesc);
+        miopenCreateTensorDescriptor(&targetGradDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+    ~MSELossDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(targetDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+
+        miopenDestroyTensorDescriptor(inputGradDesc);
+        miopenDestroyTensorDescriptor(targetGradDesc);
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+    int GetandSetData() override;
+    void GetInputTensorLengthsFromCmdLine();
+    int AllocateBuffersAndCopy() override;
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyBackward() override;
+    int VerifyForward() override;
+
+private:
+    InputFlags inflags;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t targetDesc;
+    miopenTensorDescriptor_t outputDesc;
+
+    miopenTensorDescriptor_t inputGradDesc;
+    miopenTensorDescriptor_t targetGradDesc;
+
+    std::unique_ptr<GPUMem> input_buf;
+    std::unique_ptr<GPUMem> target_buf;
+    std::unique_ptr<GPUMem> output_buf;
+    std::unique_ptr<GPUMem> workspace_buf;
+
+    std::unique_ptr<GPUMem> input_grad_buf;
+    std::unique_ptr<GPUMem> target_grad_buf;
+
+    std::vector<Tgpu> input;
+    std::vector<Tgpu> target;
+    std::vector<Tgpu> output;
+
+    std::vector<Tgpu> input_grad;
+    std::vector<Tgpu> target_grad;
+
+    std::vector<Tref> output_host;
+    std::vector<Tref> input_grad_host;
+    std::vector<Tref> target_grad_host;
+
+    float divisor;
+};
+
+template <typename Tgpu, typename Tref>
+int MSELossDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward Cat (Default=1)", "int");
+    inflags.AddInputFlag("contiguous", 'Z', "0", "Use Contiguous Tensors", "int");
+    inflags.AddTensorFlag("in_tensors", 'I', "1", "Input Tensors");
+    inflags.AddInputFlag("divisor", 'D', "1", "Divisor", "float");
+    inflags.AddInputFlag("reduction", 'r', "custom", "Reduction", "string");
```

Comment:
Please indicate "none", "sum", "mean" in the reduction of MIOpenDriver mseloss --help.

Reply:
I’ve added the descriptions